dingo/gw/domains/uniform_frequency_domain.py

```diff
@@ -0,0 +1,255 @@
+        data = data[tuple(sl)]
+
+        # Set data value below f_min to low_value.
+        sl[axis] = slice(0, self.min_idx)
+        data[tuple(sl)] = low_value
+
+        return data
+
+    def get_sample_frequencies_astype(self, data):
+        """
+        Returns a 1D frequency array compatible with the last index of data array.
+
+        Decides whether array is numpy or torch tensor (and cuda vs cpu), and whether it
+        contains the leading zeros below f_min.
+
+        Parameters
+        ----------
+        data : Union[np.array, torch.Tensor]
+            Sample data
+
+        Returns
+        -------
+        frequency array compatible with last index
+        """
+        f = super().get_sample_frequencies_astype(data)
+
+        # Whether to include zeros below f_min
+        if data.shape[-1] == len(self) - self.min_idx:
+            f = f[self.min_idx :]
+        elif data.shape[-1] != len(self):
+            raise TypeError(
+                f"Data with {data.shape[-1]} frequency bins is "
+                f"incompatible with domain."
+            )
+
+        return f
+
+    def __len__(self):
+        """Number of frequency bins in the domain [0, f_max]"""
+        return int(self.f_max / self.delta_f) + 1
+
+    @property
+    def sample_frequencies(self) -> np.ndarray:
+        if self._sample_frequencies is None:
+            num_bins = len(self)
+            self._sample_frequencies = np.linspace(
+                0.0, self.f_max, num=num_bins, endpoint=True, dtype=np.float32
+            )
+        return self._sample_frequencies
+
+    @property
+    def frequency_mask(self) -> np.ndarray:
+        """Mask which selects frequency bins greater than or equal to the
+        starting frequency"""
+        if self._frequency_mask is None:
+            self._frequency_mask = self.sample_frequencies >= self.f_min
+        return self._frequency_mask
+
+    def _reset_caches(self):
+        self._sample_frequencies = None
+        self._sample_frequencies_torch = None
+        self._sample_frequencies_torch_cuda = None
+        self._frequency_mask = None
+
+    @property
+    def frequency_mask_length(self) -> int:
+        """Number of samples in the subdomain domain[frequency_mask]."""
+        mask = self.frequency_mask
+        return len(np.flatnonzero(np.asarray(mask)))
+
+    @property
+    def min_idx(self):
+        return round(self._f_min / self._delta_f)
+
+    @property
+    def max_idx(self):
+        return round(self._f_max / self._delta_f)
+
+    @property
+    def window_factor(self):
+        return self._window_factor
+
+    @window_factor.setter
+    def window_factor(self, value):
+        """Set self._window_factor."""
+        self._window_factor = float(value)
+
+    @property
+    def f_max(self) -> float:
+        """The maximum frequency [Hz] is typically set to half the sampling
+        rate."""
+        return self._f_max
+
+    @f_max.setter
+    def f_max(self, value):
```

Comment:
```suggestion
    def f_max(self, value: float | np.ndarray):
```
Or can this only be a float?

Reply:
float